iosHyperskillApp/iosHyperskillApp/Sources/Modules/Streak/Views/Modals/StreakRecoveryModalView.swift

```diff
@@ -0,0 +1,96 @@
+import SwiftUI
+
+extension StreakRecoveryModalView {
+    struct Appearance {
+        let largeSpacing: CGFloat = 32
+        let gemsBadgeWidthHeight: CGFloat = 36
+    }
+}
+
+struct StreakRecoveryModalView: View {
+    private(set) var appearance = Appearance()
+
+    let recoveryPriceAmount: String
+    let recoveryPriceLabel: String
+    let modalText: String
+
+    weak var streakRecoveryModalDelegate: StreakRecoveryModalDelegate?
+
+    var body: some View {
+        VStack(alignment: .leading, spacing: appearance.largeSpacing) {
+            Image(Images.Profile.Streak.RecoverModal.fire)
+
+            VStack(alignment: .leading, spacing: LayoutInsets.largeInset) {
+                Text(Strings.Streak.RecoverModal.title)
+                    .foregroundColor(.primaryText)
+                    .font(.title3)
+                    .bold()
+
+                Text(modalText)
+                    .foregroundColor(.primaryText)
+                    .font(.body)
+
+                HStack(alignment: .center, spacing: LayoutInsets.smallInset) {
+                    Image(Images.StepQuiz.ProblemOfDaySolvedModal.gemsBadge)
+                        .renderingMode(.original)
+                        .resizable()
+                        .aspectRatio(contentMode: .fit)
+                        .frame(widthHeight: appearance.gemsBadgeWidthHeight)
+
+                    Text(recoveryPriceAmount)
+                        .foregroundColor(.primaryText)
+                        .font(.body)
+                        .bold()
+
+                    Text(recoveryPriceLabel)
+                        .foregroundColor(.primaryText)
+                        .font(.body)
+                }
+            }
+
+            VStack(alignment: .leading, spacing: LayoutInsets.smallInset) {
+                Text(Strings.Streak.RecoverModal.warning)
+                    .foregroundColor(.secondaryText)
+                    .font(.body)
+
+                Button(Strings.Streak.RecoverModal.restoreStreak) {
+                    streakRecoveryModalDelegate?.streakRecoveryModalDidTapRestoreStreakButton()
+                }
+                .buttonStyle(RoundedRectangleButtonStyle(style: .violet))
+
+                Button(Strings.Streak.RecoverModal.noThanks) {
+                    streakRecoveryModalDelegate?.streakRecoveryModalDidTapNoThanksButton()
+                }
+                .buttonStyle(OutlineButtonStyle())
+            }
+        }
+        .padding([.horizontal, .bottom])
+        .onAppear {
+            streakRecoveryModalDelegate?.streakRecoveryModalDidAppear()
+        }
+        .onDisappear {
+            streakRecoveryModalDelegate?.streakRecoveryModalDidDisappear()
+        }
```

Comment:
Let's don't use SwiftUI callbacks, they work not as we expected. That's why we have `UIViewControllerEventsWrapper`.

I propose to move this to the `StreakRecoveryModalViewController`.
See for example `StageImplementUnsupportedModalViewController` how we are handling those callbacks.